torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -1538,7 +1560,21 @@ void Reducer::finalize_backward() {
         ? detail::parseCppCommHookResult(bucket.future_work->value())
         : comm_hook_->parseHookResult(bucket.future_work->value());
     if (bucket.expect_sparse_gradient) {
-      bucket.gradients.copy_(future_result);
+      if (sparse_metadata_ && bucket.global_unique_id.has_value()) {
```

Comment:
is it possible for sparse_metadata = true and bucket.global_unique_id.has_value = False? In what cases can that happen?

Reply:
there shouldn't be a case, so yeah the check seems redundant